lua: statwatcher test fails on home ubuntu 19.10

```
PASS: lua/t1004-statwatcher.t 1 - require 'flux'
PASS: lua/t1004-statwatcher.t 2 - create new flux handle
PASS: lua/t1004-statwatcher.t 3 - error is nil
PASS: lua/t1004-statwatcher.t 4 - Created statwatcher
PASS: lua/t1004-statwatcher.t 5 - got current stat ok
PASS: lua/t1004-statwatcher.t 6 - got prev stat ok
PASS: lua/t1004-statwatcher.t 7 - opened file
PASS: lua/t1004-statwatcher.t 8 - line 1 matches
PASS: lua/t1004-statwatcher.t 9 - got current stat ok
PASS: lua/t1004-statwatcher.t 10 - got prev stat ok
FAIL: lua/t1004-statwatcher.t 11 - reactor exited normally: No such file or directory
ERROR: lua/t1004-statwatcher.t - exited with status 1
============================================================================
Testsuite summary for flux-core 0.13.0-174-gb7ffd16a1
============================================================================
# TOTAL: 12
# PASS:  10
# SKIP:  0
# XFAIL: 0
# FAIL:  1
# XPASS: 0
# ERROR: 1
============================================================================
See t/test-suite.log
============================================================================
make[2]: *** [Makefile:2579: test-suite.log] Error 1
make[2]: Leaving directory '/data/hd1/Repositories/flux-framework/flux-core/t'
make[1]: *** [Makefile:2687: check-TESTS] Error 2
make[1]: Leaving directory '/data/hd1/Repositories/flux-framework/flux-core/t'
make: *** [Makefile:2768: check-am] Error 2

→ lua -v       
Lua 5.1.5  Copyright (C) 1994-2012 Lua.org, PUC-Rio
```

I think the `no such file or directory` error is due to the test timing out (it takes about 5 seconds to fail).  The file does exist in the trash-directory, and contains the expected output:
```
→ cat trash-directory.lua-t1004-statwatcher/xyzzy
Hello
Goodbye


→ wc -l trash-directory.lua-t1004-statwatcher/xyzzy
3 trash-directory.lua-t1004-statwatcher/xyzzy
```

Note that only 1 line is confirmed to match (not all 3), and that the callback is indeed firing twice (as evidenced by the `got ... stat ok`. I believe that the issue is related to the file not "appearing" to have new lines from the perspective of the stat watcher callback in the second call.  If tweak the test to close and re-open the file with every call to the callback, then it works just fine. (diff below)

```diff
diff --git a/t/lua/t1004-statwatcher.t b/t/lua/t1004-statwatcher.t
index d06ab9e92..3be1eac0e 100755
--- a/t/lua/t1004-statwatcher.t
+++ b/t/lua/t1004-statwatcher.t
@@ -29,10 +29,12 @@ local s, err = f:statwatcher {
     handler = function (s, st, prev)
         ok (st, "got current stat ok")
         ok (prev, "got prev stat ok")
-        if not finfo.fp then
-            finfo.fp = io.open (file, "r")
-            ok (finfo.fp, "opened file")
+        if finfo.fp then
+           finfo.fp:close ()
         end
+        finfo.fp = io.open (file, "r")
+        ok (finfo.fp, "opened file")
+        finfo.line = 0
         for line in finfo.fp:lines () do
             finfo.line = finfo.line + 1
             is (line, expected [finfo.line], "line "..finfo.line.." matches")
```

I'm not sure if this is the right fix, because I'm not sure I fully understand the root cause.

As an aside, I'm unsure how to get a different `reason` for the `flux_reactor_stop_error` in Lua. I tried setting `posix.errno (posix.ETIMEDOUT)`, but that didn't seem to work.  The Lua bindings are grabbing the reason off the stack, but I'm not sure how to get a different reason on the stack in the first place.  I think it would be nice for this timeout failure to return `ETIMEDOUT` rather than `ENOENT`.

--- a/t/lua/t1004-statwatcher.t
+++ b/t/lua/t1004-statwatcher.t
@@ -15,7 +15,7 @@ is (err, nil, "error is nil")
 -- Test timeout: 5s
 f:timer {
     timeout = 5000,
-    handler = function () f:reactor_stop_error () end
+    handler = function () f:reactor_stop_error ("test timed out") end
 }
 
 -- Create statwatcher for file "xyzzy"

--- a/t/lua/t1004-statwatcher.t
+++ b/t/lua/t1004-statwatcher.t
@@ -29,10 +29,9 @@ local s, err = f:statwatcher {
     handler = function (s, st, prev)
         ok (st, "got current stat ok")
         ok (prev, "got prev stat ok")
-        if not finfo.fp then
-            finfo.fp = io.open (file, "r")
-            ok (finfo.fp, "opened file")
-        end
+        finfo.fp = io.open (file, "r")
+        ok (finfo.fp, "opened file")
+        finfo.line = 0
         for line in finfo.fp:lines () do
             finfo.line = finfo.line + 1
             is (line, expected [finfo.line], "line "..finfo.line.." matches")
@@ -40,6 +39,8 @@ local s, err = f:statwatcher {
                 f:reactor_stop ()
             end
         end
+        finfo.fp:close()
+        finfo.fp = nil
     end
 }
 ok (s, "Created statwatcher")
@@ -67,7 +68,7 @@ local r, err = f:reactor ()
 ok (r, "reactor exited normally: "..tostring (err))
 
 -- Close fp so we do not carry a file reference into done_testing()
-finfo.fp:close ()
+if finfo.fp then finfo.fp:close () end
 
 done_testing ()
 

